src/spikeinterface/preprocessing/silence_periods.py

```diff
@@ -79,9 +78,7 @@ def __init__(self, recording, list_periods, mode="zeros", **random_chunk_kwargs)
             rec_segment = SilencedPeriodsRecordingSegment(parent_segment, periods, mode, noise_levels)
             self.add_recording_segment(rec_segment)
 
-        self._kwargs = dict(
-            recording=recording.to_dict(), list_periods=list_periods, mode=mode, noise_levels=noise_levels
-        )
+        self._kwargs = dict(recording=recording, list_periods=list_periods, mode=mode, noise_levels=noise_levels)
```

Comment:
```suggestion
        self._kwargs = dict(recording=recording, list_periods=list_periods, mode=mode, noise_levels=noise_levels)
```
@yger I don't think this fixes the issue. Actually the `self.noise_levels` in the `SilenceRecording` is not used.
I think that you should add `noise_levels=None` to the `__init__`, so that if it's already computed it is not recomputed.

Accordingly, this should change:
```
if mode in ["noise"]:
    noise_levels = get_noise_levels(recording, return_scaled=False, concatenated=True, **random_chunk_kwargs)
else:
    noise_levels = None
```
to:
```
if mode in ["noise"]:
    if noise_levels is None:
        noise_levels = get_noise_levels(recording, return_scaled=False, concatenated=True, **random_chunk_kwargs)
else:
    noise_levels = None
```

Reply:
I  made the changes requested, but the code was already working, even with multiple jobs. Note that I thought the noise_levels are cached now. So if they have been already computed, they might be reloaded no? Aren't they saved as properties of the recording?

Reply:
That is true! Then we should have remove it from the `self._kwargs`, since caching/reloading would be done directly by the `get_noise_levels`. No?

Reply:
Yes, but I think that numerous function are requesting noise_levels in the init/kwargs, no? Should it be a more general refactoring? This could be discussed